### PR TITLE
Added version locking for protobuf extensions in genproto.sh

### PR DESCRIPTION
**Pull request by geekodour**
Fixes: #1702 

Signed-off-by: Hrishikesh Barman <hrishikeshbman@gmail.com>

### CODE REVIEW

**Comment by mxinden** [comment]
@simonpasquier would you mind looking into this with your experience through https://github.com/prometheus/prometheus/pull/5075?

**Comment by geekodour** [comment]
@simonpasquier I am having some issues understanding the rationale behind using `tools.go`. then installing it using `go install` I'll look into this again tomorrow.

the I do:
(the commit id is the exact version from the current script in this PR)
```
$ go get github.com/gogo/protobuf/protoc-gen-gogo@971cbfd2e72b513a28c74af7462aee0800248d69 
```
then the following is added into `go.mod`
```
github.com/gogo/protobuf v0.0.0-20171123125729-971cbfd2e72b
```

can you look into it?

**Comment by simonpasquier** [comment]
The idea behind `tools.go` is that it will the module recorded in `go.mod`. Otherwise running `make unused` will remove it. You can also notice that `go install` is used with the `-mod=vendor` flag which uses the files vendored with the project.

**Comment by simonpasquier** [comment]
@geekodour yes any change to the `vendor/` directory should be committed as well. The idea is that AlertManager's builds are reproducible even when the external dependencies are unavailable or tampered (though this last case shouldn't happen with Go modules).

**Comment by geekodour** [comment]
checking the failed test.

**Comment by simonpasquier** [comment]
Last thing: please add a step to Circle CI that would run `make proto` to ensure that the generated code always matches with the protobuf definition. See https://github.com/prometheus/prometheus/blob/24f19f03dbcb8791837b5d4e95827c180c319497/.circleci/config.yml#L19-L26

File: .circleci/config.yml [hunk 1]
```diff
@@ -19,6 +19,14 @@ jobs:
     - setup_remote_docker
     - run: make promu
     - run: make
+    - run:
+       command: |
+         curl -s -L https://github.com/protocolbuffers/protobuf/releases/download/v3.5.1/protoc-3.5.1-linux-x86_64.zip > /tmp/protoc.zip
```
Comment (mxinden) [inline .circleci/config.yml]: Great, thanks for also adding this!

**Pull request merged**